packages/flutter/lib/src/painting/decoration_image.dart

```diff
@@ -317,6 +321,25 @@ class DecorationImagePainter {
   }
 }
 
+/// Used by [paintImage] to report image sizes drawn at the end of the frame.
+List<ImageSizeInfo> _pendingImageSizeInfo = <ImageSizeInfo>[];
```

Comment:
should we use a `Set<ImageSizeInfo>` 
or 
`Map<ImageSizeInfo, Size>`
as we don't really care whether an image was rendered 10 times or 1 time.. We just care what the largest size it was rendered at was. I would expect the # of unique images should be significantly lower than the total # of issues for cases like using an image as a background texture uses multiple places in the UI. I don't have data though.